Fallback for cli if esbuild output not present.

> Is this the error that you were seeing generated when calling './bin/bids-validator'?
> `Error: Cannot find module 'bids-validator/cli'`
> 
> I'm seeing that error when running './bin/bids-validator' in both this branch and master at the moment.

That's because the entry point only runs the built version (when installed from npm or built locally with `npm run build`) so a separate issue. We should have a fallback for that but it's a separate issue.

_Originally posted by @nellh in https://github.com/bids-standard/bids-validator/issues/1324#issuecomment-918266959_

--- a/bids-validator/bin/bids-validator
+++ b/bids-validator/bin/bids-validator
@@ -1,9 +1,20 @@
 #!/usr/bin/env node
-try {
-  const { default: cli } = require('bids-validator/cli')
+function entry(cli) {
   cli(process.argv.slice(2)).catch(code => {
     process.exit(code)
   })
+}
+
+try {
+  const { default: cli } = require('bids-validator/cli')
+  entry(cli)
 } catch (err) {
-  console.log(err)
+  if (err.code === 'MODULE_NOT_FOUND') {
+    // For dev, use esbuild-runner if not testing a build
+    require('esbuild-runner/register')
+    const { default: cli } = require('../cli.js')
+    entry(cli)
+  } else {
+    console.log(err)
+  }
 }

--- a/bids-validator/package.json
+++ b/bids-validator/package.json
@@ -60,6 +60,7 @@
     "chai": "",
     "codecov": "^3.1.0",
     "esbuild": "^0.11.5",
+    "esbuild-runner": "^2.2.1",
     "esbuild-plugin-globals": "^0.1.1",
     "eslint": "^5.2.0",
     "eslint-config-prettier": "^2.9.0",

